### ISSUE TITLE
collection.distinct("field") query fails with a MongoCommandException when collection does not exists

### ISSUE THREAD

**Issue by aurelien-minvielle** [bwaldvogel/mongo-java-server#178]
Hello,

Using a `distinct(...)` query on an absent collection throws a MongoCommandException stating the collection does not exist.

Here a sample failing unit test : 

```
public class SimpleTest {

    private MongoCollection<org.bson.Document> collection;
    private MongoClient client;
    private MongoServer server;

    @BeforeEach
    public void setUp() {
        server = new MongoServer(new MemoryBackend());

        // bind on a random local port
        InetSocketAddress serverAddress = server.bind();

        client = new MongoClient(new ServerAddress(serverAddress));
        MongoDatabase database = client.getDatabase("testdb");
        collection = database.getCollection("testcollection");
    }

    @AfterEach
    public void tearDown() {
        client.close();
        server.shutdown();
    }

    @Test
    public void testDistinctQueryOnAbsentCollection() {
        var distinct = collection.distinct("foo", String.class);
        assertThat(distinct.cursor().hasNext()).isFalse();
    }
}
```

Result:
```
com.mongodb.MongoCommandException: Command failed with error -1: 'Collection [testdb.testcollection] not found.' on server localhost:57506. The full response is {"$err": "Collection [testdb.testcollection] not found.", "errmsg": "Collection [testdb.testcollection] not found.", "ok": 0}

	at com.mongodb.internal.connection.ProtocolHelper.getCommandFailureException(ProtocolHelper.java:175)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:302)
	at com.mongodb.internal.connection.InternalStreamConnection.sendAndReceive(InternalStreamConnection.java:258)
	at com.mongodb.internal.connection.UsageTrackingInternalConnection.sendAndReceive(UsageTrackingInternalConnection.java:99)
	at com.mongodb.internal.connection.DefaultConnectionPool$PooledConnection.sendAndReceive(DefaultConnectionPool.java:450)
	at com.mongodb.internal.connection.CommandProtocolImpl.execute(CommandProtocolImpl.java:72)
	at com.mongodb.internal.connection.DefaultServer$DefaultServerProtocolExecutor.execute(DefaultServer.java:226)
	at com.mongodb.internal.connection.DefaultServerConnection.executeProtocol(DefaultServerConnection.java:269)
	at com.mongodb.internal.connection.DefaultServerConnection.command(DefaultServerConnection.java:131)
	at com.mongodb.internal.connection.DefaultServerConnection.command(DefaultServerConnection.java:123)
	at com.mongodb.operation.CommandOperationHelper.executeCommand(CommandOperationHelper.java:343)
	at com.mongodb.operation.CommandOperationHelper.executeCommand(CommandOperationHelper.java:334)
	at com.mongodb.operation.CommandOperationHelper.executeCommandWithConnection(CommandOperationHelper.java:220)
	at com.mongodb.operation.CommandOperationHelper$5.call(CommandOperationHelper.java:206)
	at com.mongodb.operation.OperationHelper.withReadConnectionSource(OperationHelper.java:463)
	at com.mongodb.operation.CommandOperationHelper.executeCommand(CommandOperationHelper.java:203)
	at com.mongodb.operation.DistinctOperation.execute(DistinctOperation.java:183)
	at com.mongodb.operation.DistinctOperation.execute(DistinctOperation.java:62)
	at com.mongodb.client.internal.MongoClientDelegate$DelegateOperationExecutor.execute(MongoClientDelegate.java:194)
	at com.mongodb.client.internal.MongoIterableImpl.execute(MongoIterableImpl.java:143)
	at com.mongodb.client.internal.MongoIterableImpl.iterator(MongoIterableImpl.java:92)
	at com.mongodb.client.internal.MongoIterableImpl.cursor(MongoIterableImpl.java:97)
```

I understand the expected behaviour to be to return an empty result instead.

That's the behaviour I get on mongodb server v3.4, v3.6 and v4.0 using either the Java driver or the mongo shell :
```
> use testdb
switched to db testdb
> db.testcollection.distinct("a-field")
[ ]
> 
```

**Comment by bwaldvogel** [comment]
Thanks for the good bug report. This issue is fixed with eb84de922267d42db9871f3d10e0243df05f797a.

**Comment by bwaldvogel** [comment]
The fix is available in version `1.38.0`.
